### DISCUSSION TITLE
Please offer a way to hide the menu bar icon

### DISCUSSION THREAD

**Discussion by JamesHenry** [johnlindquist/kit#497] [Ideas]
Hey @johnlindquist,

I'm fortunate to be one of the lucky folks setting up their new 2021 MacBook Pro, and have a related feature request.

Now that this model has the notch with the menu bar wrapping around it, menu bar real estate is at an all-time premium. I, like I'm sure many do, exclusively use the keyboard shortcut to trigger the Kit prompt, so the current menu bar icon offers no value to me and I would love to be able to free up that precious space for other things.

Further data point: from what I can see Script Kit is currently the only app I have installed that does not offer a way to hide its menu bar icon, so _hopefully_ it's relatively straightforward to support in a similar way.

Kind regards,

James

**Comment by johnlindquist** [comment]
I _think_ it will be fairly straightforward. I'll try to get it in tomorrow's build.

How's the new laptop treating you?

**Comment by JamesHenry** [comment]
It’s incredible! I’m jumping all the way from the 2016 13 inch (butterfly keyboard, Touch Bar, no ports and all the other complaints) so everything about it is crazy better.

**Comment by johnlindquist** [comment]
Got this in today's build:

![CleanShot 2021-11-05 at 08 50 36](https://user-images.githubusercontent.com/36073/140530046-1c2290a4-a87d-490c-b616-968eb10cf034.png)

**Reply by JamesHenry** [reply]
May I suggest actually changing this to Menu Bar?

Apple itself does not appear to use "system tray" and this impacts discoverability I think. Apple consistently refers to it as Menu Bar, and searching for system tray in settings yields no results (but you can see Dock and Menu Bar in the screenshot below):

![image](https://user-images.githubusercontent.com/900523/141146427-e3d85ba7-d68c-4121-ae65-b5e9ee79590f.png)

**Comment by JamesHenry** [comment]
Works perfectly, thank you @johnlindquist!

**Comment by JamesHenry** [comment]
Actually sorry @johnlindquist, it doesn't seem to persist the preference across quitting the app and restarting, please could we add that in too? (Plus my suggested rename above) 🙏

**Reply by johnlindquist** [reply]
Makes sense. I just used "tray" because that's what it's called in the API.

I just realized I stored the preference, but forgot to read it when the app launched 🤦‍♂️ I'll get that fixed

**Reply by JamesHenry** [reply]
Haha classic Apple - definitely makes sense why you picked that then. Is it possible to make it so that searching for either term shows the result within the Kit UI? Then my point around discoverability is resolved regardless of whether the name officially changes

And cool, glad it's not too tricky to enable to preference restoration

**Reply by johnlindquist** [reply]
Ok, this should all be sorted out in the latest update

**Reply by JamesHenry** [reply]
Awesome, thanks for continuing to iterate on this, it works great! 🙏

**Comment by noahrezentes** [comment]
Hi @johnlindquist, wondering if this is still possible with current version? No luck searching either term in the prompt.

**Reply by johnlindquist** [reply]
In ~/.kenv/.env set:

KIT_TRAY=false
